src/native.rs

```diff
@@ -2123,6 +2124,74 @@ impl HasContext for Context {
         gl.GetQueryObjectuiv(query, parameter, &mut value);
         value
     }
+
+    unsafe fn create_transform_feedback(&self) -> Result<Self::TransformFeedback, String> {
+        let gl = &self.raw;
+        let mut name = 0;
+        gl.GenTransformFeedbacks(1, &mut name);
+        Ok(name)
+    }
+
+    unsafe fn delete_transform_feedback(&self, transform_feedback: Self::TransformFeedback) {
+        let gl = &self.raw;
+        gl.DeleteTransformFeedbacks(1, &transform_feedback);
+    }
+
+    unsafe fn bind_transform_feedback(&self, target: u32, transform_feedback: Option<Self::TransformFeedback>) {
+        let gl = &self.raw;
+        gl.BindTransformFeedback(target, transform_feedback.unwrap_or(0));
+    }
+
+    unsafe fn begin_transform_feedback(&self, primitive_mode: u32) {
+        let gl = &self.raw;
+        gl.BeginTransformFeedback(primitive_mode);
+    }
+
+    unsafe fn end_transform_feedback(&self) {
+        let gl = &self.raw;
+        gl.EndTransformFeedback();
+    }
+
+    unsafe fn pause_transform_feedback(&self) {
+        let gl = &self.raw;
+        gl.PauseTransformFeedback();
+    }
+
+    unsafe fn resume_transform_feedback(&self) {
+        let gl = &self.raw;
+        gl.ResumeTransformFeedback();
+    }
+
+    unsafe fn transform_feedback_varyings(&self, program: Self::Program, varyings: &[&str], buffer_mode: u32) {
+        let gl = &self.raw;
+        gl.TransformFeedbackVaryings(
+            program,
+            varyings.len() as i32,
+            varyings.as_ptr() as *const *const native_gl::types::GLchar,
```

Comment:
Please double check this!

Reply:
To have properly null-terminated strings here, we will need to convert each `&str` into a `CString`, then use [`.as_ptr()` on that instead](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.as_ptr) (note the warning about lifetimes). There is a response in https://users.rust-lang.org/t/converting-str-to-const-c-char/23115 which explains this a bit.